data/libs/Commodities.lua

```diff
@@ -0,0 +1,63 @@
+-- Copyright © 2008-2020 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local Economy = require 'Economy'
+
+local commodities = Economy.GetCommodities()
+local economies = Economy.GetEconomies()
+
+local EquipType = _G.EquipType
+local cargo = {}
+
+local CARGOLANGRESOURCE = "commodity"
+
+-- TODO: normalize icon name conventions; ideally we don't need this table at all and can simply
+-- use the commodity name for the icon
+local icon_names = {
+	hydrogen = "Hydrogen",
+	liquid_oxygen = "Liquid_Oxygen",
+	water = "Water",
+	carbon_ore = "Carbon_ore",
+	metal_ore = "Metal_ore",
+	metal_alloys = "Metal_alloys",
+	precious_metals = "Precious_metals",
+	plastics = "Plastics",
+	fruit_and_veg = "Fruit_and_Veg",
+	animal_meat = "Animal_Meat",
+	live_animals = "Live_Animals",
+	liquor = "Liquor",
+	grain = "Grain",
+	slaves = "Slaves",
+	textiles = "Textiles",
+	fertilizer = "Fertilizer",
+	medicines = "Medicines",
+	consumer_goods = "Consumer_goods",
+	computers = "Computers",
+	rubbish = "Rubbish",
+	radioactives = "Radioactive_waste",
+	narcotics = "Narcotics",
+	nerve_gas = "Nerve_Gas",
+	military_fuel = "Military_fuel",
+	robots = "Robots",
+	hand_weapons = "Hand_weapons",
+	air_processors = "Air_processors",
+	farm_machinery = "Farm_machinery",
+	mining_machinery = "Mining_machinery",
+	battle_weapons = "Battle_weapons",
+	industrial_machinery = "Industrial_machinery"
+}
+
+-- TODO: don't create a separate EquipType for each commoditity,
+-- instead store cargo separately from ship equipment
```

Comment:
Ah, that old can of worms... I hadn't the patience of doing so back then as we had at least one prominent mod (Scout+) that would have had a lot of breakage over this. How is the mod situation nowadays ?

Reply:
In my eyes, Pioneer mods don't exist right now. I haven't seen a recent mod (other than maybe one personal hobby project) be announced for the entire time I've been contributing, and all of the mods I can find are at least 4 years old. I'd rather move forward and break old mods than stay in this limbo of massively inflated savefiles and awkward cargo handling. (No offense meant to you, the equipment system is a massive step up from what was previous! It's just... not a good fit for cargo.)

If someone wants to upgrade existing mods, absolutely more power to them and I'd be more than happy to support them, but until that time I'm not going to let old and inactive mods be an excuse for leaving things as the status quo.